obspy/io/mseed/core.py

```diff
@@ -785,6 +785,16 @@ def _write_mseed(stream, filename, encoding=None, reclen=None, byteorder=None,
                 trace_attr['encoding'] = 1
             elif trace.data.dtype.type == np.dtype(native_str('|S1')).type:
                 trace_attr['encoding'] = 0
+            #int64 data not supported; if possible, downcast to int32, otherwise create error message
+            elif trace.data.dtype.type == np.int64:
+                #check if data can be safely downcast to int32
+                trace_data1 = trace.data.copy().astype(np.int32)
```

Comment:
I think this is copying the data in memory twice, according to numpy docs `trace.data.astype(np.int32, copy=True)` should be fine and only copy once then

Reply:
Oh, I see you were doing it the same as in other parts of the code below, and I see now why it's done like this. We still support numpy 1.6.1 which has `ndarray.astype(t)` (https://docs.scipy.org/doc/numpy-1.6.0/reference/generated/numpy.ndarray.astype.html?highlight=astype#numpy.ndarray.astype) and no `copy=..` kwarg.

@krischer I think we should avoid copying data twice in the whole mseed writer code, so I think the way to go would be to define a little helper function up top of the file and replace all these double copying operations with it? You agree?

```
def _np_copy_astype(data, dtype):
    try:
        return data.astype(dtype, copy=True)
    except TypeError:
        return data.copy().astype(dtype)
```